packages/devtools_shared/lib/src/utils/license_utils.dart

```diff
@@ -0,0 +1,240 @@
+// Copyright 2024 The Flutter Authors
```

Comment:
can we put this code under `devtools/tool` since `devtools_shared` is a public package shipped on pub? That way we don't have to worry about breaking changes for a tool utility like this. If at some point we want to pull this code out and of the DevTools repo and put it in a shared package, we can do so.

Reply:
@kenzieschmoll Sure, but can you point me to where the unit tests are run for tool? It wasn't obvious to me.

Reply:
It appears we don't have any. You can create a new folder `tool/test` and add the test there.

Then I'd recommend adding a new github actions job to test the `tool/` directory. To do this:
1. Create a new shell script `tool/ci/tool_tests.sh`:
```sh
#!/bin/bash

# Copyright 2024 The Chromium Authors. All rights reserved.
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.

# Fast fail the script on failures.
set -ex

source ./tool/ci/setup.sh

pushd $DEVTOOLS_DIR/tool
echo `pwd`
flutter test test/
popd
```
2. Create a new job in the github actions setup (see [test-packages](https://github.com/flutter/devtools/blob/master/.github/workflows/build.yaml/#L77) for an example):
```yaml
  test-tool:
    name: tool test
    needs: flutter-prep
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
    steps:
      - name: git clone
        uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
      - name: Load Cached Flutter SDK
        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9
        with:
          path: |
            ./tool/flutter-sdk
          key: flutter-sdk-${{ runner.os }}-${{ needs.flutter-prep.outputs.latest_flutter_candidate }}
      - name: tool/ci/tool_tests.sh
        run: ./tool/ci/tool_tests.sh
```